packages/expo-router/src/getRoutes.ts

```diff
@@ -383,6 +384,8 @@ export function assertDuplicateRoutes(filenames: string[]) {
 
 /** Given a Metro context module, return an array of nested routes. */
 export function getRoutes(contextModule: RequireContext, options?: Options): RouteNode | null {
+  options = { importMode: EXPO_ROUTER_IMPORT_MODE, ...options };
```

Comment:
Typed routes need to force `async` loading, so this option can now be overridden.